src/com/android/volley/Request.java

```diff
@@ -69,10 +69,12 @@
      * Supported request methods.
      */
     public enum ReturnStrategy {
-        DOUBLE, // You'll get called back into twice, once for cached result and once for network delivery
         NETWORK_IF_NO_CACHE, // You'll get the network response only if the cache misses
         CACHE_IF_NETWORK_FAILS, // You'll get the cached response only if the network errors / fails
-        NETWORK_ONLY // Skip the cache
+        NETWORK_ONLY, // Skip the cache
+        // END LEGACY RETURN STRATEGIES
+        DOUBLE, // You'll get called back into up to twice, once for cached result and once for network delivery
+        SINGLE // guarantees that response will be called only once
```

Comment:
needs more description here; all the other return types besides double are singe-response themselves.